### PR TITLE
 Mapping Parser And Batch Check model

### PR DESCRIPTION
**Pull request by daveaugustus**
### :nut_and_bolt: Description: What code changed, and why?
Map Config from HaDeployConfig

### :chains: Related Resources
https://chefio.atlassian.net/jira/software/c/projects/CHEF/boards/482?selectedIssue=CHEF-5612

### :+1: Definition of Done
- Test cases need to be written
- Mapping needs to be added for s3 and gcs
- Edge cases need to be checked
- Pipeline must pass
### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/automate-cli/pkg/verifyserver/models/batchcheck.go [hunk 1]
```diff
@@ -271,16 +277,42 @@ func (c *Config) populateCommonConfig(haConfig *config.HaDeployConfig) error {
...
 	objectStorageConfig := haConfig.GetObjectStorageConfig()
-	c.Backup = &Backup{
-		ObjectStorage: &ObjectStorage{
-			BucketName: objectStorageConfig.BucketName,
-			AWSRegion:  objectStorageConfig.Region,
-			AccessKey:  objectStorageConfig.AccessKey,
-			SecretKey:  objectStorageConfig.SecretKey,
-			Endpoint:   objectStorageConfig.Endpoint,
-		},
+	if objectStorageConfig.Location == AWS_S3 {
```
Comment (YashviJain01) [inline components/automate-cli/pkg/verifyserver/models/batchcheck.go:285]: Are we handling the case where location is empty and set to s3?

File: lib/config/hadeployconfigverify_test.go [hunk 2]
```diff
@@ -106,6 +106,7 @@ func TestVerifyObjectStorage(t *testing.T) {
 		{
 			name: "Valid Object Storage Configuration",
 			objectStorage: &ConfigObjectStorage{
+				Location:   "s3",
```
Comment (YashviJain01) [inline lib/config/hadeployconfigverify_test.go:109]: Have we added test case for any successful case for gcs?

File: lib/config/hadeployconfigverify_test.go [hunk 3]
```diff
@@ -157,6 +162,7 @@ func TestVerifyObjectStorage(t *testing.T) {
 		{
 			name: "Invalid Region",
 			objectStorage: &ConfigObjectStorage{
+				Location:   "s3",
```
Comment (YashviJain01) [inline lib/config/hadeployconfigverify_test.go:165]: This should support the blank value, where location is not set.

File: lib/config/hadeployconfigverify.go [hunk 4]
```diff
@@ -198,6 +198,22 @@ func (c *HaDeployConfig) verifyObjectStorage(objectStorage *ConfigObjectStorage)
 			}
 		}
 	}
+	if objectStorage.Location == AWS_S3 {
```
Comment (Dmaddu) [inline lib/config/hadeployconfigverify.go:201]: These are already available from line number 185

File: lib/config/testdata/gcsservicefile.json [hunk 5]
```diff
@@ -10,4 +10,4 @@
   "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
   "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/test",
   "universe_domain": "main"
-}
\ No newline at end of file
+}
```
Comment (Dmaddu) [inline lib/config/testdata/gcsservicefile.json:13]: There are no changes in this file, please remove this file from PR

File: components/automate-cli/pkg/verifyserver/models/batchcheck.go [hunk 6]
```diff
@@ -7,10 +7,17 @@ import (
...
 	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/sirupsen/logrus"
+
 	"github.com/chef/automate/lib/config"
 	"github.com/gofiber/fiber/v2"
 )
 
+const (
+	AWS_S3            = "s3"
+	GCP_CLOUD_STORAGE = "gcp"
```
Comment (Dmaddu) [inline components/automate-cli/pkg/verifyserver/models/batchcheck.go:18]: Are we using GCP or GCS, please decide on one. In the previous PR, the constant was GCS, and in this its GCP. As per standalone it should be GCS

File: components/automate-cli/pkg/verifyserver/models/batchcheck.go [hunk 7]
```diff
@@ -271,15 +278,46 @@ func (c *Config) populateCommonConfig(haConfig *config.HaDeployConfig) error {
...
+		c.Backup = &Backup{
+			ObjectStorage: &ObjectStorage{
+				BucketName: objectStorageConfig.BucketName,
+				AWSRegion:  objectStorageConfig.Region,
+				AccessKey:  objectStorageConfig.AccessKey,
+				SecretKey:  objectStorageConfig.SecretKey,
+				Endpoint:   objectStorageConfig.Endpoint,
+			},
+		}
+	} else if objectStorageConfig.Location == GCP_CLOUD_STORAGE {
```
Comment (Dmaddu) [inline components/automate-cli/pkg/verifyserver/models/batchcheck.go:302]: Do we have any validation if the provided value is neither s3 nor gcs?

**Review by Dmaddu — CHANGES_REQUESTED** [review]

File: components/automate-cli/pkg/verifyserver/models/batchcheck.go [hunk 8]
```diff
@@ -271,15 +278,49 @@ func (c *Config) populateCommonConfig(haConfig *config.HaDeployConfig) error {
...
+					AuthURI:                 objectStorageConfig.GcpServiceAccount.AuthURI,
+					TokenURI:                objectStorageConfig.GcpServiceAccount.TokenURI,
+					AuthProviderX509CertURL: objectStorageConfig.GcpServiceAccount.AuthProviderX509CertURL,
+					ClientX509CertURL:       objectStorageConfig.GcpServiceAccount.ClientX509CertURL,
+					UniverseDomain:          objectStorageConfig.GcpServiceAccount.UniverseDomain,
+				},
+			},
+		}
+	} else {
+		logrus.Errorf("For now we don't support location: %s", objectStorageConfig.Location)
```
Comment (Dmaddu) [inline components/automate-cli/pkg/verifyserver/models/batchcheck.go:322]: plz change this message

**Review by Dmaddu — CHANGES_REQUESTED** [review]

**Comment by daveaugustus** [comment]
Pipeline:

https://buildkite.com/chef-oss/chef-automate-main-verify/builds/30647
https://buildkite.com/chef/chef-automate-main-verify-private/builds/31928

**Review by Dmaddu — APPROVED** [review]
